JDP190901-13

pawban opened the pull request:
**Encja Group - implementacja**

Utworzyłem klasę Group w pakiecie com.kodilla.ecommercee.domain.
Klasa zawiera pola odpowiadające polom z powiązanego obiektu DTO, czyli
id i name - każde opisane stosownymi adnotacjiami.
Id jest kluczem głównym generowanym automatycznie; name nie może być
nullem.
Metody clearId() używam zawszew w serwisie w metodzie tworzącej nowy
obiekt, aby upewnić się, że rzeczywiście zostanie utworzony nowy obiekt,
a nie będzie modyfikowany istniejący (w przypadku, gdy przesłany obiekt
zawiera pole id).

pawban committed:
Encja Group - implementacja

Utworzyłem klasę Group w pakiecie com.kodilla.ecommercee.domain.
Klasa zawiera pola odpowiadające polom z powiązanego obiektu DTO, czyli
id i name - każde opisane stosownymi adnotacjiami.
Id jest kluczem głównym generowanym automatycznie; name nie może być
nullem.
Metody clearId() używam zawszew w serwisie w metodzie tworzącej nowy
obiekt, aby upewnić się, że rzeczywiście zostanie utworzony nowy obiekt,
a nie będzie modyfikowany istniejący (w przypadku, gdy przesłany obiekt
zawiera pole id).

pawban committed:
Revert "Encja Group - implementacja"

This reverts commit 077846d9

Arelias commented:
CircleCi wyrzuca bledy do metod get, jak pisalem rowniez na grupie, zapewne przez brak     annotationProcessor 'org.projectlombok:lombok' w twoim pull requescie z gradle

pawban committed:
Encja Group - implementacja (poprawione)

Utworzyłem klasę Group w pakiecie com.kodilla.ecommercee.domain.
Klasa zawiera pola odpowiadające polom z powiązanego obiektu DTO, czyli
id i name - każde opisane stosownymi adnotacjiami.
Id jest kluczem głównym generowanym automatycznie; name nie może być
nullem.
Metody clearId() używam zawszew w serwisie w metodzie tworzącej nowy
obiekt, aby upewnić się, że rzeczywiście zostanie utworzony nowy obiekt,
a nie będzie modyfikowany istniejący (w przypadku, gdy przesłany obiekt
zawiera pole id).

pawban commented:
Zepsułem pierwszy commit (wrzuciłem niedokończone pliki niedotyczące tego taska) - dlatego nie udało się zbudować projektu. Wycofałem zepsuty commit i zrobiłem go ponownie - tym razem już jest wszystko ok.

pawban committed:
Encja Group - implementacja repozytorium

Utworzyłem repozytorium dla encji Group.
Nie nadpisywałem metod dziedziczonych z JpaRepository, ponieważ nie
wydaje mi się to sensowne. Kontroler w obecnym kształcie nie wymaga
moim zdaniem żadnych metod ponad te dostępne standardowo
w interfejsie JpaRepository.

pawban commented:
Utworzyłem repozytorium dla encji Group.
Nie nadpisywałem metod dziedziczonych z JpaRepository, ponieważ nie wydaje mi się to konieczne. Kontroler w obecnym kształcie nie wymaga moim zdaniem żadnych metod ponad te dostępne standardowo w interfejsie JpaRepository.

pawban committed:
Group - implementacja encji (poprawiona nazwa tabeli)

Nazwa tabeli w bazie danych, w której będą przechowywne encje, została
zmieniona na 'product_group', ponieważ domyślna nazwa - 'group' -
powodowała błąd składani SQL ('group' jest słowem kluczowym SQL
i nie może być użyte jako nazwa tabeli).

pawban committed:
Merge branch 'master' into JDP190901-13
